flink-core/src/main/java/org/apache/flink/api/dag/Transformation.java

```diff
@@ -273,7 +273,8 @@ public ResourceSpec getPreferredResources() {
      * @param managedMemoryUseCase The use case that this transformation declares needing managed
      *     memory for.
      * @param weight Use-case-specific weights for this transformation. Used for sharing managed
-     *     memory across transformations for OPERATOR scope use cases.
+     *     memory across transformations for OPERATOR scope use cases. For consistency, the APIs
+     *     declare their weights as a kibibyte value.
```

Comment:
I'm not entirely sure that 1kb is a good unit for weights in all the use cases. E.g., our internal Gemini state backend also uses managed memory in a per-operator bias, which sets the weight to be number of states the operator maintains.

Not saying we should change things for a special internal use case. Admittedly, having a consistent unit for all use cases is not causing problems at the moment, because there's currently only one operator scope managed memory use case. But thinking of future flexibility, maybe we should avoid having unnecessarily strong assumptions. In this case, what we really need is a consistent unit of weight for one specific managed memory use case, rather than all use cases.

In particular, I'd suggest to add the 1kb definition of weight to the JavaDoc of `ManagedMemoryUseCase#OPERATOR`, and add a pointer here to remind the callers to check the weight definition of the declared use case.

Reply:
@xintongsong I'm also not happy with this PR. It is only the minimal solution to make DataStream API batch mode compatible with Table API's batch mode. A KB unit also has the downside of a potential overflow if the value is set to high. Also casting `MemorySize.getKibiBytes()` to `int` is not very nice. Shall we change this to MB instead?

Reply:
I'm ok with either KB or MB.
- For KB, it needs 2^31 total weights to get an overflow. That's roughly 2PB, per slot, which is not likely the case.
- For MB, I also don't see much chances that an operator needs to tune its memory as fine-grained as less than 1MB.